sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -135,4 +137,35 @@ export class ApplicationController extends BaseController {
 
     return assessment;
   }
+
+  /**
+   * Confirm Assessment of a Student.
+   * @param applicationId application id to be updated.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Patch(":applicationId/confirm-assessment")
+  async studentConfirmAssessment(
```

Comment:
I think about one more scenario, in order for the confirmation of the assessment to happen, should the record be in some specific state?
I believe that we need to introduce a check to verify if the application is in the expected state to have the assessment confirmed. Does it make sense @abschwenker?

Reply:
I think application_status to be in 'Assessment'. Is that what you meant?

Reply:
I am really not sure. I just believe that the student cannot change the status at any time and the API must restrict it.
What you mentioned makes sense to me, but I was looking to get some business knowledge from @abschwenker about it.

Reply:
For a student to confirm their assessment the application must be in application_status="Assessment". Does that answer the question?